sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncClient.java

```diff
@@ -63,31 +52,25 @@
     public static final String DEFAULT_CONSUMER_GROUP_NAME = "$Default";
 
     private static final String RECEIVER_ENTITY_PATH_FORMAT = "%s/ConsumerGroups/%s/Partitions/%s";
-    private static final String SENDER_ENTITY_PATH_FORMAT = "%s/Partitions/%s";
 
     private final ClientLogger logger = new ClientLogger(EventHubAsyncClient.class);
     private final MessageSerializer messageSerializer;
-    private final Mono<EventHubConnection> connectionMono;
+    private final EventHubLinkProvider linkProvider;
     private final AtomicBoolean hasConnection = new AtomicBoolean(false);
     private final ConnectionOptions connectionOptions;
     private final String eventHubName;
-    private final EventHubProducerOptions defaultProducerOptions;
     private final EventHubConsumerOptions defaultConsumerOptions;
     private final TracerProvider tracerProvider;
 
     EventHubAsyncClient(ConnectionOptions connectionOptions, TracerProvider tracerProvider,
-                        MessageSerializer messageSerializer, Mono<EventHubConnection> eventHubConnectionMono) {
+                        MessageSerializer messageSerializer, EventHubLinkProvider linkProvider) {
 
         this.connectionOptions = Objects.requireNonNull(connectionOptions, "'connectionOptions' cannot be null.");
         this.tracerProvider = Objects.requireNonNull(tracerProvider, "'tracerProvider' cannot be null.");
         this.eventHubName = connectionOptions.getEntityPath();
```

Comment:
This client should also have a method to retrieve `fullyQualifiedNamespace`

Reply:
Fixed